### PR TITLE
[SEP-31] Delegate Kyc flow to SEP-12 using `transaction_id`

### PR DESCRIPTION
**Pull request by JiahuiWho**
As title

### CODE REVIEW

File: ecosystem/sep-0012.md [hunk 1]
```diff
@@ -145,17 +145,19 @@ protocols, such as [SEP-6](sep-0006.md) and [SEP-31](sep-0031.md), that use SEP-
...
 example, a customer could have an `ACCEPTED` status for the `small-transaction-amount` `type` parameter but have a
-`NEEDS_INFO` status for the `large-transaction-amount` type. Therefore it is recommended to always pass the `type`
-parameter when making requests to `GET /customer` or `PUT /customer`, even though the field is optional to accomodate
+`NEEDS_INFO` status for the `large-transaction-amount` type. Therefore, it is recommended to always pass the `type`
+parameter when making requests to `GET /customer` or `PUT /customer`, even though the field is optional to accommodate
 for implementations that do not require `type`.
 
 If the implementor requires the same set of fields for all customers, there is no need for the `type` parameter.
 
+When the `type` parameter is used with `transaction_id`, it can only be `sep6`, `sep31-sender`, or `sep31-receiver`,
```
Comment (Ifropc) [inline ecosystem/sep-0012.md:158]: Should we mark it mandatory if used with transaction id?

File: ecosystem/sep-0012.md [hunk 2]
```diff
@@ -145,17 +145,19 @@ protocols, such as [SEP-6](sep-0006.md) and [SEP-31](sep-0031.md), that use SEP-
...
 example, a customer could have an `ACCEPTED` status for the `small-transaction-amount` `type` parameter but have a
-`NEEDS_INFO` status for the `large-transaction-amount` type. Therefore it is recommended to always pass the `type`
-parameter when making requests to `GET /customer` or `PUT /customer`, even though the field is optional to accomodate
+`NEEDS_INFO` status for the `large-transaction-amount` type. Therefore, it is recommended to always pass the `type`
+parameter when making requests to `GET /customer` or `PUT /customer`, even though the field is optional to accommodate
 for implementations that do not require `type`.
 
 If the implementor requires the same set of fields for all customers, there is no need for the `type` parameter.
 
+If `transaction_id` is in use, `type` must be provided and set to either `sep6`, `sep31-sender`, or `sep31-receiver`,
```
Comment (philipliu) [inline ecosystem/sep-0012.md:158]:
```suggestion
If `transaction_id` is in use, `type` must be provided and set (e.g. `sep6`, `sep31-sender`, or `sep31-receiver`)
```

We let SEP-31 anchors define its types, we shouldn't restrict it here.

File: ecosystem/sep-0031.md [hunk 3]
```diff
@@ -857,9 +853,16 @@ In certain cases the Receiving Anchor might need to request updated information
...
 until the Sending Anchor makes another `SEP-12 PUT /customer` request to update. The Sending Anchor can check which
-fields need to be updated by making a `SEP-12 GET /customer` request including the `id` or `account` & `memo`
-parameters. The Receiving Anchor should respond with a `NEEDS_INFO` status and `last_name` included in the fields
-described.
+fields need to be updated by making a `SEP-12 GET /customer` request. The Receiving Anchor should respond with a
+`NEEDS_INFO` status, a `provided_fields` including all provided fields, and a `fields` including all fields need to be
+updated .
+
+1. `GET /transaction` and verify transaction status is `pending_customer_info_update`
+2. Get required fields via [SEP-12] `GET /customer` request. It is recommended to use the combination of
```
Comment (philipliu) [inline ecosystem/sep-0031.md:861]: Is the brackets around `SEP-12` meant to link to the protocol?

File: ecosystem/sep-0031.md [hunk 4]
```diff
@@ -857,9 +853,16 @@ In certain cases the Receiving Anchor might need to request updated information
...
 Since this information was sent via SEP-12, the transaction should go into the `pending_customer_info_update` state
 until the Sending Anchor makes another `SEP-12 PUT /customer` request to update. The Sending Anchor can check which
-fields need to be updated by making a `SEP-12 GET /customer` request including the `id` or `account` & `memo`
-parameters. The Receiving Anchor should respond with a `NEEDS_INFO` status and `last_name` included in the fields
-described.
+fields need to be updated by making a `SEP-12 GET /customer` request. The Receiving Anchor should respond with a
+`NEEDS_INFO` status, a `provided_fields` including all provided fields, and a `fields` including all fields need to be
+updated .
+
+1. `GET /transaction` and verify transaction status is `pending_customer_info_update`
```
Comment (philipliu) [inline ecosystem/sep-0031.md:860]: nit: missing period at the end.

File: ecosystem/sep-0031.md [hunk 5]
```diff
@@ -857,9 +853,16 @@ In certain cases the Receiving Anchor might need to request updated information
...
+2. Get required fields via [SEP-12] `GET /customer` request. It is recommended to use the combination of
+   `transaction_id` and customer type `type` (`sep31-sender` and `sep31-receiver`) as part of the request.
```
Comment (philipliu) [inline ecosystem/sep-0031.md:862]: Should we mention `sep31-sender` and `sep31-receiver` here? It looks like these two types are the only acceptable types.

Comment (JiahuiWho) [reply ecosystem/sep-0031.md]: Will remove

**Review by Ifropc — APPROVED** [review]

**Pull request merged**